test/E2ETests/Smoke.E2E.Tests/TestDataSourceExtensibilityTests.cs

```diff
@@ -19,5 +19,116 @@ public void ExecuteTestDataSourceExtensibilityTests()
                 "CustomTestDataSourceTestMethod1 (1,2,3)",
                 "CustomTestDataSourceTestMethod1 (4,5,6)");
         }
+
+        /* This test needs a reference to "Microsoft.TestPlatform.AdapterUtilities" and "Microsoft.TestPlatform.ObjectModel" NuGet packages.
+         * It's used to debug FQN changes in test discovery and test execution.
```

Comment:
Did you intend to leave the below code commented?

Reply:
Yes, it requires a lot of new NuGet packages and changes to that project. I'll move it somewhere else or create another project for these tests. Right now, I am leaving it commented since this work is currently ongoing. 

Ultimately, I'll enable these tests with upcoming PRs.